### PR TITLE
feat: remove duplicate tick labels from axis

### PR DESCRIPTION
**Pull request by rshen91**
## Summary

Closes https://github.com/elastic/elastic-charts/issues/445 and addresses related kibana issue https://github.com/elastic/kibana/issues/47045

**BREAKING CHANGE**: 
`showDuplicateTicks` is set to false by default - this changes current behavior as seen in https://github.com/elastic/kibana/issues/47045

This PR removes duplicate tickLabels for an axis instead of taking into account unique tickValues. For instance, if the tickLabel on the axis is showing days, then millisecond differences in the tickValues should not generate duplicate tickLabels. However, if the consumer wants to show duplicate tickLabels, then this can be configured with the `showDuplicateTicks` config prop but isn't set to true by default. 

There has been a story added within Axes - `Axes - Duplicate Ticks`

Mar-18-2020 13-57-55

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR. 

~~- [] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)~~
- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [x] Proper documentation or storybook story was added for features that require explanation or tutorials
- [x] Unit tests were updated or added to match the most common scenarios
- [x] Each commit follows the [convention](https://github.com/elastic/elastic-charts/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/scales/scale_continuous.ts [hunk 1]
```diff
@@ -139,6 +139,8 @@ interface ScaleOptions {
   isSingleValueHistogram: boolean;
   /** Show only integer values **/
   integersOnly?: boolean;
+  /** Show duplicate tick values default to false */
+  duplicateTicks?: boolean;
```
Comment (markov00) [inline src/scales/scale_continuous.ts]: Why we need this in the scale option? the `getDuplicateTicks` doesn't use that but use directly the AxisSpec props

File: src/chart_types/xy_chart/utils/axis_utils.ts [hunk 2]
```diff
@@ -444,14 +443,42 @@ export function getAvailableTicks(
...
+          uniqueTickLabels.push(allTicks[index]);
+          // once uniqueTickLabels has the unique label, remove the label from the labels array so it doesnt create a duplicate
+          labels.splice(i, 1);
+        }
+      }
+    });
+    return uniqueTickLabels;
+  } else {
+    return allTicks;
+  }
```
Comment (markov00) [inline src/chart_types/xy_chart/utils/axis_utils.ts]: I think this code can be simplified a bit, let's chat on that directly

File: src/scales/scale_continuous.ts (hunk 1 above)
Comment (rshen91) [reply src/scales/scale_continuous.ts]: Ah thanks for spotting this, I didn't remove this properly after enabling the feature - I'm removing this in commit 5037110

File: src/chart_types/xy_chart/utils/specs.ts [hunk 3]
```diff
@@ -525,11 +525,14 @@ export interface AxisSpec extends Spec {
   style?: AxisStyle;
   /** Show only integar values **/
   integersOnly?: boolean;
+  /** Remove duplicate ticks, default is false*/
+  duplicateTicks?: boolean;
```
[the remark this answers is not in the document]
Comment (rshen91) [reply src/chart_types/xy_chart/utils/specs.ts]: I like the naming a lot better - commit 5037110 for changes

File: stories/line/10_duplicate_ticks.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+import React from 'react';
+import { Axis, Chart, LineSeries, Position, ScaleType, niceTimeFormatter } from '../../src/';
+import { KIBANA_METRICS } from '../../src/utils/data_samples/test_dataset_kibana';
+import { boolean } from '@storybook/addon-knobs';
+import { DateTime } from 'luxon';
+
+export const example = () => {
+  const now = DateTime.fromISO('2019-01-11T00:00:00.000')
+    .setZone('utc+1')
```
[the remark this answers is not in the document]
Comment (rshen91) [reply stories/line/10_duplicate_ticks.tsx]: Weirdly, utc+1 wasn't aligning but local seems to do the trick. Not really sure how that is possible. Commit ead2e9e

File: src/chart_types/xy_chart/utils/axis_utils.test.ts [hunk 5]
```diff
@@ -1436,4 +1439,88 @@ describe('Axis computational utils', () => {
 
     expect(getAxisTickLabelPadding(axisConfigTickLabelPadding, axisSpecStyle)).toEqual(2);
   });
+  test('should show unique tick labels if duplicateTicks is set to false', () => {
+    const now = DateTime.fromISO('2019-01-11T00:00:00.000')
+      .setZone('utc+1')
+      .toMillis();
+    const oneDay = 1000 * 60 * 60 * 24;
```
[the remark this answers is not in the document]
Comment (rshen91) [reply src/chart_types/xy_chart/utils/axis_utils.test.ts]: Cool! Pls see commit 55bf6ee for changes 💪

File: src/chart_types/xy_chart/utils/axis_utils.test.ts [hunk 6]
```diff
@@ -1436,4 +1439,88 @@ describe('Axis computational utils', () => {
...
+    const formatter = niceTimeFormatter([now, now + oneDay * 31]);
```
[the remark this answers is not in the document]
Comment (rshen91) [reply src/chart_types/xy_chart/utils/axis_utils.test.ts]: Changes in commit 55bf6ee thanks!

File: src/chart_types/xy_chart/utils/axis_utils.ts [hunk 7]
```diff
@@ -444,14 +440,42 @@ export function getAvailableTicks(
 
     return [firstTick, lastTick];
   }
-  return ticks.map((tick) => {
+  return enableDuplicatedTicks(axisSpec, scale, offset, tickFormatOptions);
+}
+
+export function enableDuplicatedTicks(
```
[the remark this answers is not in the document]
Comment (rshen91) [reply src/chart_types/xy_chart/utils/axis_utils.ts]: Thanks for pointing this out - changed the config property naming in commit 
6ba6751

File: src/chart_types/xy_chart/utils/axis_utils.ts (hunk 2 above)
Comment (rshen91) [reply src/chart_types/xy_chart/utils/axis_utils.ts]: Thanks a ton for your help with me on this. I added commit 594c6f6 which I think gets done what we discussed, but lmk what you think!

**Comment by rshen91** [comment]
jenkins, test this please

**Pull request merged**